// This is the public API offered by your entity.
syntax = "proto3";

import "google/protobuf/empty.proto";
import "akkaserverless/annotations.proto";
import "google/api/annotations.proto";

import "thing_domain.proto";

package com.example.value;


message Thing {
  string thing_id = 1 [(akkaserverless.field).entity_key = true];
  string name = 2;
}

message GetThingCommand {
  string thing_id = 1 [(akkaserverless.field).entity_key = true];
}

service ThingValueService {

    rpc AddThing(Thing) returns (ThingState) {
        option (google.api.http) = {
        post: "/value/things"
            body: "*"
        };
    }

    rpc GetThing(GetThingCommand) returns (ThingState) {
        option (google.api.http) = {
        get: "/view/things/{thing_id}"
        };
    }
    
}

